storage/src/rocksdb_client.cpp

```diff
@@ -646,6 +647,121 @@ void Client::removeAllCheckpoints() const {
   }
 }
 
+Client::RocksDbStorageMetrics::RocksDbStorageMetrics(const std::vector<::rocksdb::Tickers> &tickers,
+                                                     const Client &owningClient)
+    : StorageMetricsBase({"storage_rocksdb", std::make_shared<concordMetrics::Aggregator>()},
+                         update_metrics_interval_msec_),
+      owning_client_(owningClient),
+      total_db_disk_size_(metrics_.RegisterAtomicGauge("storage_rocksdb_total_db_disk_size", 0)),
+      all_mem_tables_ram_usage_(metrics_.RegisterAtomicGauge("storage_rocksdb_mem_tables_ram_usage", 0)),
+      all_column_families_block_cache_ram_usage_(
+          metrics_.RegisterAtomicGauge("storage_rocksdb_column_families_block_cache_ram_usage", 0)),
+      indexes_and_filters_ram_usage_(metrics_.RegisterAtomicGauge("storage_rocksdb_indexes_and_filter_ram_usage", 0)),
+      rocksdb_total_ram_usage_(metrics_.RegisterAtomicGauge("storage_rocksdb_total_ram_usage", 0)) {
+  for (const auto &pair : ::rocksdb::TickersNameMap) {
+    if (std::find(tickers.begin(), tickers.end(), pair.first) != tickers.end()) {
+      auto metric_suffix = pair.second;
+      std::replace(metric_suffix.begin(), metric_suffix.end(), '.', '_');
+      active_tickers_.emplace(pair.first, metrics_.RegisterAtomicGauge("storage_" + metric_suffix, 0));
+    }
+  }
+  metrics_.Register();
+}
+
+void Client::RocksDbStorageMetrics::setMetricsDataSources(std::shared_ptr<::rocksdb::SstFileManager> sourceSstFm,
```

Comment:
It's faster to pass shared_ptr by reference.

Reply:
It's corrrect, but it really depends on the intention. If the intention is to gain shared ownerhip (and seems to me like this is the reason), I think that current code is the right one.

Reply:
it's an existing code that I just moved to a different file - didn't wanna touch it